scripts/add-license-headers.sh

```
@@ -79,73 +117,79 @@ add_ocaml_header() {
+        # header mangling).
+        if head -"$header_end_line" "$file" 2>/dev/null | grep "SPDX-FileCopyrightText:" | grep -qi "$contributor_email"; then
             # Contributor exists - check if year needs updating
-            local contributor_line=$(head -"$header_end_line" "$file" | grep "SPDX-FileCopyrightText:" | grep "$contributor_email")
+            local contributor_line=$(head -"$header_end_line" "$file" | grep "SPDX-FileCopyrightText:" | grep -i "$contributor_email")
             local existing_years=$(echo "$contributor_line" | grep -oP '\d{4}(-\d{4})?')
             local first_year=$(echo "$existing_years" | cut -d- -f1)
-            
+
             if [[ "$existing_years" =~ - ]]; then
                 # Has year range - check if last year matches
                 local end_year=$(echo "$existing_years" | cut -d- -f2)
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)